plugin/cmd/kube-scheduler/app/server.go

```diff
@@ -114,7 +114,7 @@ func Run(s *options.SchedulerServer) error {
 	rl := &resourcelock.EndpointsLock{
 		EndpointsMeta: metav1.ObjectMeta{
 			Namespace: "kube-system",
-			Name:      "kube-scheduler",
+			Name:      s.SchedulerName,
```

Comment:
I still think it is unexpected the scheduler name they pick will be used as the leader lock endpoint. What happens if they have a name that is invalid as an endpoint object name (I'd sort of expect them to have used a namespaced custom scheduler name like `acme.com/fancy-scheduler`)?